.azure-pipelines/templates/stages/deploy-stage.yml

```diff
@@ -53,10 +54,25 @@ stages:
             inputs:
               secureFile: snapcraft.cfg
           - bash: |
+              set -e
+              retry_command() {
+                # based on travis_retry.bash https://github.com/travis-ci/travis-build/blob/master/lib/travis/build/bash/travis_retry.bash
+                local result=0
+                local count=1
+                while [[ "${count}" -le 3 ]]; do
+                  # run the command in a way that doesn't disable setting `errexit`
+                  "${@}" || result="${?}"
```

Comment:
My understanding is `bash` short circuits these conditionals. As a result, I think if the command fails once, `result` will always have a nonzero value.

I think we could fix this by setting `result` equal to 0 somewhere in the loop.

Reply:
yeah wait what you're totally right, how was any of this working for travis

Reply:
oh right, they didn't have the `||`. maybe they turned off `set -e`, too.

Reply:
Maybe that's what they mean by the comment:
```
# run the command in a way that doesn't disable setting `errexit`
```
and they want `set -e` to break `travis_retry`?

That seems like a strange choice for them to make, but I don't understand that comment otherwise.

Reply:
I think they essentially do want `set -e` to break `travis_retry`. This was added in https://github.com/travis-ci/travis-build/pull/1850. Before that PR, they implemented the command similar to what you've done here.

My understanding for the reason they made this change is people were running commands like `travis_retry source some_script.sh` and if `some_script.sh` included `set -e` it would keep running even if a command failed because it was sourced into a conditional which effectively disables `set -e`. Isn't bash great? (An extra fun detail is apparently this behavior doesn't apply to `bash` on macOS.)

I don't think we should worry about this extreme edge case, but I also think the comment no longer applies to our code so we should delete it so we don't spend time in the future trying to understand it like I just did.

Otherwise, this PR LGTM!

Reply:
Thanks for looking into that!